base/stats.go

```diff
@@ -0,0 +1,791 @@
+package base
+
+import (
+	"expvar"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	StatViewFormat = "%s.%s"
+
+	NamespaceKey                 = "sgw"
+	ResourceUtilizationSubsystem = "resource_utilization"
+
+	SubsystemCacheKey           = "cache"
+	SubsystemDatabaseKey        = "database"
+	SubsystemDeltaSyncKey       = "delta_sync"
+	SubsystemGSIViews           = "gsi_views"
+	SubsystemReplication        = "replication"
+	SubsystemReplicationPull    = "replication_pull"
+	SubsystemReplicationPush    = "replication_push"
+	SubsystemSecurity           = "security"
+	SubsystemSharedBucketImport = "shared_bucket_import"
+)
+
+type SgwStats struct {
+	GlobalStats     *GlobalStat         `json:"global"`
+	DbStats         map[string]*DbStats `json:"per_db"`
+	ReplicatorStats *ReplicatorStats    `json:"per_replication,omitempty"`
+
+	mutex                     sync.Mutex
```

Comment:
Is this mutex being used?

Reply:
No. I was questioning whether I needed one for the DbStats map but came to the conclusion that I don't think its needed.